src/App.tsx

```diff
@@ -1,24 +1,414 @@
+    if (inputValue.trim() !== '') {
+      const tempTodoItem: Todo = {
+        title: inputValue,
+        userId: USER_ID,
+        completed: false,
+        id: getRandomNumber(),
+      };
+
+      try {
+        await client.post('/todos', tempTodoItem);
+        setTodo((prevTodo) => [...prevTodo, tempTodoItem]);
+      } catch (error) {
+        console.log('There is an error', error);
+      }
+    } else {
+      setErrorMessageField(true);
+    }
+
+    setInputValue('');
+  };
+
+  const deleteTodo = async (id: number) => {
+    const tempTodos = todo.filter((element) => {
+      return element.id !== id;
+    });
+
+    setTodo(tempTodos);
+
+    try {
+      await client.delete(`/todos/${id}`);
+    } catch (error) {
+      console.log('There is an issue.', error);
+      setDeleteErrorMessage('Unable to delete a todo');
+    }
+  };
+
+  const changeAll = async () => {
+    const updatedTodos = todo.map((element) => {
+      return {
+        ...element,
+        completed: !element.completed,
+      };
+    });
+
+    setTodo(updatedTodos);
+
+    try {
+      updatedTodos.map(async (todoItem) => {
+        await client.patch(`/todos/${todoItem.id}`, {
+          ...todoItem,
+          completed: todoItem.completed,
+        });
+      });
+    } catch (error) {
+      console.log('There is an issue updating todos.', error);
+    }
+  };
+
+  const setCompleted = async (id: number) => {
+    const updatedTodo = todo.map((obj) => {
+      if (obj.id === id) {
+        return {
+          ...obj,
+          completed: !obj.completed,
+        };
+      }
+
+      return obj;
+    });
+
+    const none = todo.some((element) => {
+      return element.id === id;
+    });
+
+    if (!none) {
+      setIsEditingTodoAllowed(true);
+      setErrorMessageField(true);
+      setEditTodo('Unable to loading a todo');
+      setIsThereIssue(true);
+    }
+
+    setTodo(updatedTodo);
+
+    try {
+      const todoToUpdate = todo.find((elem) => elem.id === id);
+
+      if (todoToUpdate) {
+        await client.patch(`/todos/${id}`, {
+          completed: !todoToUpdate.completed,
+          title: todoToUpdate.title,
+          userId: USER_ID,
+          id,
+        });
+      }
+    } catch (error) {
+      setDeleteErrorMessage('Unable to delete a todo');
+      console.log('There is an issue.');
+    }
+  };
+
+  const deleteCompletedTodos = async () => {
+    const completedTodoIds = todo
+      .filter((element) => element.completed)
+      .map((element) => element.id);
+
+    try {
+      await Promise.all(
+        completedTodoIds.map((id) => client.delete(`/todos/${id}`)),
+      );
+
+      setTodo((prevTodo) => prevTodo.filter((element) => !element.completed));
+    } catch (error) {
+      console.log('There is an issue deleting completed todos.', error);
+      setDeleteErrorMessage('Unable to delete completed todos');
+    }
+  };
+
+  useEffect(() => {
+    fetchTodos();
+
+    return () => {
+      clearTimeout(timeoutId);
```

Comment:
Use 'timeoutId.current' instead of 'timeoutId' to clear the timeout. Replace 'clearTimeout(timeoutId)' with 'clearTimeout(timeoutId.current)'